controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -669,3 +724,160 @@ bool wfa_ca::create_cmdu(ieee1905_1::CmduMessageTx &cmdu_tx, ieee1905_1::eMessag
+        for (size_t tlv_member_idx = 0; tlv_member_idx < tlv_members.size(); tlv_member_idx++) {
+            std::string lookup_str =
+                tlv_members[tlv_member_idx] + (tlv_idx > 0 ? std::to_string(tlv_idx) : "");
+            auto it = params.find(lookup_str);
+            if (it == params.end()) {
+                if (tlv_idx == 0 && tlv_member_idx == 0) {
+                    // Skipping TLV with no index (index 0), checking TLV index 1
+                    skip = true;
+                    break;
+                } else if (tlv_idx > 0 && tlv_member_idx == 0) {
+                    // No indexed TLVs found, finish
+                    finish = true;
+                    break;
+                } else {
+                    err_string = "missing param name '" + lookup_str + "'";
+                    return false;
+                }
+            }
+
+            // Type
+            if (tlv_member_idx == 0) {
+                tlv_hex.type = &it->second;
+
+                // Length
+            } else if (tlv_member_idx == 1) {
+                tlv_hex.length = &it->second;
+
+                // Value
+            } else if (tlv_member_idx == 2) {
+                // The hex values contains '{' and '}' delimiters, but we don't care about them.
+                it->second.erase(std::remove(it->second.begin(), it->second.end(), '{'),
+                                 it->second.end());
+                it->second.erase(std::remove(it->second.begin(), it->second.end(), '}'),
+                                 it->second.end());
+                tlv_hex.value = &it->second;
+            } else {
+                LOG(ERROR) << "Illegal tlv_member_idx value: " << int(tlv_member_idx);
+                err_string = err_internal;
+                return false;
+            }
+
+            // Validate hex notation on value or list of values separated by space
+            auto values = string_utils::str_split(it->second, ' ');
+            for (auto value : values) {
+                if (!validate_hex_notation(value)) {
+                    err_string = "param name '" + lookup_str + "' has invalid hex notation value";
+                    return false;
+                }
+
+                // Type and Length fields must have fixed size:
+
+                // Type
+                if (tlv_member_idx == 0 && value.size() != 4) {
+                    err_string = "type field is too large: " + value;
+                    return false;
+
+                    // Length
+                } else if (tlv_member_idx == 1 && value.size() != 6) {
+                    err_string = "length field is too large: " + value;
+                    return false;
+                }
+            }
+        }
+
+        if (finish) {
+            return true;
+        }
+
+        tlv_idx++;
+
+        if (skip) {
+            skip = false;
+            continue;
+        }
+
+        tlv_hex_list.push_back(tlv_hex);
+
+    } while (tlv_idx < max_tlv);
```

Comment:
Is it worth to have this explicit stop condition? We can just make it an endless loop, no? There is no way that it can really be an endless loop because we stop as soon as one of the parameters is missing.

Reply:
I think it could happen. I didn't see any limitations on TLVs number.
Besides, it is harmless, and generally good practice to make a stop condition, and also prevent from static code analysis tools to complain about potential endless loop like klocwork which we use on Intel.

Reply:
> static code analysis tools to complain about potential endless loop

Indeed, that's a good reason to do this.